File: kmp/log4agent/src/commonMain/kotlin/dev/log4agent/Log4AgentEndpoint.kt
package dev.log4agent

object Log4AgentEndpoint {
    const val DEFAULT_PORT: Int = 3100
    const val DEFAULT_PATH: String = "/logs"

    fun default(): String = defaultLog4AgentEndpoint()

    fun fromHost(
        host: String,
        port: Int = DEFAULT_PORT,
        path: String = DEFAULT_PATH,
        scheme: String = "http",
    ): String {
        val normalizedPath = if (path.startsWith("/")) path else "/$path"
        return "$scheme://$host:$port$normalizedPath"
    }
}

File: ios/Sources/Log4Agent/Log4AgentEndpoint.swift
import Foundation

public enum Log4AgentEndpoint {
    public static let defaultURL = fromHost("127.0.0.1")

    public static func fromHost(
        _ host: String,
        port: Int = 3100,
        path: String = "/logs",
        scheme: String = "http"
    ) -> URL {
        var components = URLComponents()
        components.scheme = scheme
        components.host = host
        components.port = port
        components.path = path.hasPrefix("/") ? path : "/\(path)"
        return components.url!
    }
}

